src/main/java/net/sf/jabref/model/DuplicateCheck.java

```diff
@@ -63,28 +66,28 @@
      * @return boolean
      */
     public static boolean isDuplicate(BibEntry one, BibEntry two, BibDatabaseMode bibDatabaseMode) {
-
-        // First check if they are of the same type - a necessary condition:
+        // same type is mandatory
         if (!one.getType().equals(two.getType())) {
             return false;
         }
-        EntryType type = EntryTypes.getTypeOrDefault(one.getType(), bibDatabaseMode);
 
-        // The check if they have the same required fields:
-        List<String> var = type.getRequiredFieldsFlat();
+        // check for equal required fields
+        EntryType entryType = EntryTypes.getTypeOrDefault(one.getType(), bibDatabaseMode);
+        List<String> requiredFields = entryType.getRequiredFieldsFlat();
         double[] req;
-        if (var == null) {
+        if (requiredFields == null) {
```

Comment:
Are we really returning null? I think, we ommitted null, thus also line 79 is not covered by tests at all.